lib/syntax-definition/src/Language/Symbol/Operator/Prec/Poset.hs

```diff
@@ -0,0 +1,175 @@
+module Language.Symbol.Operator.Prec.Poset where
```

Comment:
Given this is fully general, you could pull it out into the `lib/` folder and make it available to other portions of the compiler that need it.